src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -531,25 +580,82 @@ class RequiredMandatoryAttribute:
                     success = False
 
         # validate there are no attributes in the range between standard and global
+        # This is de-facto already covered in the check above, assuming the spec hasn't defined any values in this range, but we should make sure
         for endpoint_id, endpoint in self.endpoints_tlv.items():
             for cluster_id, cluster in endpoint.items():
-                bad_range_values = [a for a in cluster[ATTRIBUTE_LIST_ID] if a > standard_range_max and a < global_range_min]
+                bad_range_values = [a for a in cluster[ATTRIBUTE_LIST_ID] if a >
+                                    attribute_standard_range_max and a < global_range_min]
                 for bad in bad_range_values:
                     location = AttributePathLocation(endpoint_id=endpoint_id, cluster_id=cluster_id, attribute_id=bad)
                     self.record_error(self.get_test_name(), location=location,
                                       problem=f"Attribute in undefined range {bad} in cluster {cluster_id}", spec_location=f"Cluster {cluster_id}")
                     success = False
 
+        command_standard_range_max = 0x0000_00FF
+        # Command lists only have a scoped range, so we only need to check for known command ids, no global range check
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            for cluster_id, cluster in endpoint.items():
+                if cluster_id not in chip.clusters.ClusterObjects.ALL_CLUSTERS:
+                    continue
+                standard_accepted_commands = [a for a in cluster[ACCEPTED_COMMAND_LIST_ID] if a <= command_standard_range_max]
+                standard_generated_commands = [a for a in cluster[GENERATED_COMMAND_LIST_ID] if a <= command_standard_range_max]
+                if cluster_id in chip.clusters.ClusterObjects.ALL_ACCEPTED_COMMANDS:
+                    allowed_accepted_commands = [a for a in chip.clusters.ClusterObjects.ALL_ACCEPTED_COMMANDS[cluster_id]]
+                else:
+                    allowed_accepted_commands = []
+                if cluster_id in chip.clusters.ClusterObjects.ALL_GENERATED_COMMANDS:
+                    allowed_generated_commands = [a for a in chip.clusters.ClusterObjects.ALL_GENERATED_COMMANDS[cluster_id]]
+                else:
+                    allowed_generated_commands = []
+
+                unexpected_accepted_commands = sorted(list(set(standard_accepted_commands) - set(allowed_accepted_commands)))
+                unexpected_generated_commands = sorted(list(set(standard_generated_commands) - set(allowed_generated_commands)))
```

Comment:
allowed_accepted_command within the standard scope ?

Reply:
Added a comment.